# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.cpp
# Commit: 478ac1179fa6
# Full Hash: 478ac1179fa606aa4de5cdbf6a8a11a377e139bb
# Author: Cameron McCormack <cam@mcc.id.au>
# Date: 2019-06-07 09:47:19
# Regressor Bug: 1490792
# File Overlap Count: 1
# Description:
#   Bug 1490792 - Part 1: Move font type determination into SanitizeOpenTypeData. r=jfkthame
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D33894
# ==============================================================================

diff -r 91643c6f9a42 -r 478ac1179fa6 gfx/thebes/gfxUserFontSet.cpp
--- a/gfx/thebes/gfxUserFontSet.cpp	Thu Jun 06 19:18:16 2019 +0000
+++ b/gfx/thebes/gfxUserFontSet.cpp	Thu Jun 06 18:23:27 2019 +0000
@@ -253,7 +253,10 @@
 // Returns a newly-allocated block, or nullptr in case of fatal errors.
 const uint8_t* gfxUserFontEntry::SanitizeOpenTypeData(
     const uint8_t* aData, uint32_t aLength, uint32_t& aSaneLength,
-    gfxUserFontType aFontType) {
+    gfxUserFontType& aFontType) {
+  aFontType = gfxFontUtils::DetermineFontDataType(aData, aLength);
+  Telemetry::Accumulate(Telemetry::WEBFONT_FONTTYPE, uint32_t(aFontType));
+
   if (aFontType == GFX_USERFONT_UNKNOWN) {
     aSaneLength = 0;
     return nullptr;
@@ -672,25 +675,13 @@
                    mFontDataLoadingState < LOADING_FAILED,
                "attempting to load a font that has either completed or failed");
 
-  gfxFontEntry* fe = nullptr;
-
-  gfxUserFontType fontType =
-      gfxFontUtils::DetermineFontDataType(aFontData, aLength);
-  Telemetry::Accumulate(Telemetry::WEBFONT_FONTTYPE, uint32_t(fontType));
-
   // Unwrap/decompress/sanitize or otherwise munge the downloaded data
   // to make a usable sfnt structure.
 
-  // Because platform font activation code may replace the name table
-  // in the font with a synthetic one, we save the original name so that
-  // it can be reported via the InspectorUtils API.
-  nsAutoCString originalFullName;
-
   // Call the OTS sanitizer; this will also decode WOFF to sfnt
   // if necessary. The original data in aFontData is left unchanged.
   uint32_t saneLen;
-  uint32_t fontCompressionRatio = 0;
-  size_t computedSize = 0;
+  gfxUserFontType fontType;
   const uint8_t* saneData =
       SanitizeOpenTypeData(aFontData, aLength, saneLen, fontType);
   if (!saneData) {
@@ -706,6 +697,16 @@
       saneData = nullptr;
     }
   }
+
+  // Because platform font activation code may replace the name table
+  // in the font with a synthetic one, we save the original name so that
+  // it can be reported via the InspectorUtils API.
+  nsAutoCString originalFullName;
+
+  gfxFontEntry* fe = nullptr;
+  uint32_t fontCompressionRatio = 0;
+  size_t computedSize = 0;
+
   if (saneData) {
     if (saneLen) {
       fontCompressionRatio = uint32_t(100.0 * aLength / saneLen + 0.5);